e2e/node_config.go

```diff
@@ -668,8 +668,11 @@ func baseTemplateLinux(t *testing.T, location string, k8sVersion string, arch st
 
 // this been crafted with a lot of trial and pain, some values are not needed, but it takes a lot of time to figure out which ones.
 // and we hope to move on to a different config, so I don't want to invest any more time in this-
+// please keep the kubernetesVersion in sync with componets.json so that during e2e no extra binaries are required.
 func baseTemplateWindows(t *testing.T, location string) *datamodel.NodeBootstrappingConfiguration {
-	kubernetesVersion := "1.29.9"
+	kubernetesVersion := "1.30.12"
```

Comment:
Could we parse components.json to get the latest k8s version? How is this determined for the linux tests?

Prob out of scope for your PR tho.